Some minor TD weapon yamls restructuring

reaperrr opened the pull request:
Saving a few more yaml lines and increasing the use of inheritance a bit more, to make potential future across-the-board changes easier.

reaperrr committed:
Move TD Vulcan up for more consistent order

reaperrr committed:
Move TD Grenade to largecaliber.yaml

This is more about making maintenance a little easier than logic. The file might get renamed to 'ballistic.yaml' at some point, though.

reaperrr committed:
Move TD TiberiumExplosion to explosions.yaml

reaperrr committed:
Make TD Chemspray inherit ^FlameWeapon

Results in a small reduction of yaml lines and makes it easier to make
across-the-board changes in the future.

reaperrr commented:
Updated commit order and PR name/description a bit.

This PR's code changes are relatively simple and a dependency for similar yaml changes in the other mods, so it would be nice if this doesn't go un-reviewed for too long.

pchote commented:
> it would be nice if this doesn't go un-reviewed for too long.

Unfortunately I don't see a way around this.  The business with the project logo has left a bad taste in my mouth, so i'm taking time away from OpenRA.  @obrakmann, @abcdefg30, etc are also taking time off, so we're down to basically no reviewers for now.

reaperrr commented:
Well, I assume that means the next release won't happen soon, either, so I guess it doesn't really matter.

GraionDilach commented:
I am still around sporadically, although I had no time to review this week due to my night shift and AS and the fact I got a GPU for Christmas.

:0.5: to this PR because it seems okay but I can't test it atm.

reaperrr commented:
Rebased.

reaperrr commented:
While it probably was mostly due to the general lack of active reviewers, I still have split the code changes (+ a small RA example case) to #12582 in a bid to get at least the code changes reviewed and merged a bit more quickly, as they aren't that large and and you won't have to review and test as many yaml changes there.

This PR is still pretty 'safe' to merge once the code changes are confirmed to work and merged, though.

pchote commented:
Reiterating my comment from #12582, I think it would be much better if the warhead could work this out for itself without having to manually set the search radius to 0.

reaperrr commented:
> Reiterating my comment from #12582, I think it would be much better if the warhead could work this out for itself without having to manually set the search radius to 0.

Valid point.

However, instead of closing this altogether, I just removed the code and warhead commits and updated this to only contain the unproblematic restructuring commits. We may still benefit from them if we want to make across-the-board yaml changes in other cases.

abcdefg30 commented on mods/cnc/weapons/explosions.yaml:
```diff
@@ -99,3 +99,17 @@ Napalm.Crate:
 		SmudgeType: Scorch
+
+TiberiumExplosion:
+	Inherits: ^DamagingExplosion
```
You probably want to inherit `^DamagingExplosionHE` instead. Otherwise you're changing the `Versus` values.

reaperrr replied:
Right, well spotted.

reaperrr committed:
Rename TD largecaliber.yaml to ballistics.yaml

Mostly because Grenade was added to it.

reaperrr commented:
Fixed, also renamed largecaliber.yaml to ballistics.yaml (due to Grenade, mostly).

abcdefg30 commented:
👍
